File: employee-management-ui/src/models/Employee.ts
export const Gender = {
  Male: 0,
  Female: 1,
  Other: 2,
} as const;

export type Gender = (typeof Gender)[keyof typeof Gender];

export interface Employee {
  employeeId: number;
  name: string;
  email: string;
  salary: number;
  gender: Gender;
  dateOfBirth: string;
  isActive: boolean;
  departmentId: number;
  departmentName: string;
}

File: EmployeeManagementAPI/Models/Employee.cs
using EmployeeManagementAPI.Models;

namespace EmployeeManagementAPI.Models
{
    public class Employee
    {
        public int EmployeeId { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public decimal Salary { get; set; }
        public Gender Gender { get; set; }
        public DateTime DateOfBirth { get; set; }
        public bool IsActive { get; set; }
        public int DepartmentId { get; set; }
        public Department? Department { get; set; }
    }
}
